warehouse/templates/pages/help.html

```diff
@@ -145,12 +164,26 @@ <h3 id="verified-email">{{ verified_email() }}</h3>
-          <p>If you receive an error message saying that "This password appears in a breach or has been compromised and cannot be used", you should change it all other places that you use it as soon as possible.</p>
-          <p>If you have received this error while attempting to log in or upload to PyPI, then your password has been reset and you cannot log in to PyPI until you <a href="{{ request.route_url('accounts.request-password-reset') }}">reset your password</a>.</p>
+        <p>PyPI itself has not suffered a breach. This is a protective measure to reduce the risk of <a href="https://www.owasp.org/index.php/Credential_stuffing" target="_blank" rel="noopener">credential stuffing</a> attacks against PyPI and its users.</p>
+        <p>Each time a user supplies a password — while registering, authenticating, or updating their password — PyPI securely checks whether that password has appeared in public data breaches.</p>
+        <p>During each of these processes, PyPI generates a SHA-1 hash of the supplied password and uses the first five (5) characters of the hash to check the <a href="https://haveibeenpwned.com/API/v2#SearchingPwnedPasswordsByRange" target="_blank" rel="noopener">Have I Been Pwned API</a> and determine if the password has been previously compromised. The plaintext password is never stored by PyPI or submitted to the Have I Been Pwned API.</p>
+        <p>PyPI will not allow such passwords to be used when setting a password at registration or updating your password.</p>
+        <p>If you receive an error message saying that "This password appears in a breach or has been compromised and cannot be used", you should change it all other places that you use it as soon as possible.</p>
+        <p>If you have received this error while attempting to log in or upload to PyPI, then your password has been reset and you cannot log in to PyPI until you <a href="{{ request.route_url('accounts.request-password-reset') }}">reset your password</a>.</p>
+
+        <h3 id="2fa">{{ twoFA() }}</h3>
+        <p>Users who have chosen to set up two factor authentication (2FA) on their PyPI account must provide a second method of identity verification (other than their username and password) to log in.</p>
+        <p>PyPI currently supports a single 2FA method: <a href="#totp">Generating a code through a TOTP application</a>.
+
+        <h3 id="totp">{{ totp() }}</h3>
+        <p>When enabling two factor authentication (2FA) via <a href="https://en.wikipedia.org/wiki/Time-based_One-time_Password_algorithm">TOTP</a> in your account admin, you were asked to provision an application (usually a mobile phone app) in order to generate authentication codes. Popular applications include:</p>
+        <ul>
+          <li><a href="https://freeotp.github.io/">FreeOTP</a> (open source)</li>
+          <li><a href="https://play.google.com/store/apps/details?id=org.liberty.android.freeotpplus&hl=en">FreeOTP+</a> (open source)</li>
```

Comment:
I believe it's better to not include language query param in the link.
```suggestion
          <li><a href="https://play.google.com/store/apps/details?id=org.liberty.android.freeotpplus">FreeOTP+</a> (open source)</li>
```